src/functional/ffront/past_to_itir.py

```diff
@@ -44,7 +45,12 @@ class ProgramLowering(NodeTranslator):
     ...    fieldop(inp, out=out)
     >>>
     >>> parsed = ProgramParser.apply_to_function(program)
-    >>> lowered = ProgramLowering.apply(parsed)
+    >>> fieldop_def = ir.FunctionDefinition(
+    ...     id="fieldop",
+    ...     params=[ir.Sym(id="inp")],
+    ...     expr=ir.FunCall(fun=ir.SymRef(id="deref"), args=[ir.SymRef(id="inp")])
```

Comment:
To be consistent I think `fieldop` above should then also return `inp`.

Reply:
No: in the lowered code (`lowered` variable below), the stencil in the closure is just a symbol reference to fieldop. Thus, fieldop has to be a stencil, that is, take an iterator and return a value, (otherwise the generated IR is not valid) and the ‘identity’ stencil still has to deref its input.